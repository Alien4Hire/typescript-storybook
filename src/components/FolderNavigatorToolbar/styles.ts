import { Box, Button } from '@mui/material';

import { styled } from '@mui/material/styles';

export const Divider = styled('div')(({ theme }) => ({
  borderRight: `1px solid ${theme.palette.gray.medium}`,
  height: '34px',
  margin: 'auto',
}));

export const Label = styled('div')(({ theme }) => ({
  fontWeight: '600',
  fontSize: '15px',
  letterSpacing: '0.4px',
  color: theme.palette.gray.veryDark,
  fontFamily: "'Source Sans Pro', sans-serif",
}));

export const Wrapper = styled('div')(({ theme }) => ({
  position: 'relative',
  background: theme.palette.gray.veryLight,
  paddingLeft: '56px',
}));

export const FolderForm = styled('div')`
  display: flex;
  align-items: center;
`;

export const FolderAdd = styled(Box, {
  shouldForwardProp: (prop: string) => prop !== 'active',
})(({ active, theme }) => ({
  position: 'absolute',
  top: 0,
  bottom: 0,
  left: 0,
  right: 0,
  width: 'auto',
  padding: '2px',
  backgroundColor: theme.palette?.gray.veryLight,
  display: 'none',
  ...(active && {
    display: 'flex',
    alignItems: 'center',
  }),
}));

export const CsvModalBackdrop = styled(Box)({
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: '#f1f1f1',
  borderRadius: '16px',
  padding: '32px 32px 70px',
});

export const UploadListItem = styled(Box)(({ theme }) => ({
  display: 'flex',
  alignItems: 'center',
  padding: '16px 0px',
  borderBottom: `1px solid ${theme.palette.gray.light}`,
}));

export const UploadButton = styled(Button)({
  textTransform: 'none',
  width: 'fit-content',
  marginTop: '16px',
});

export const DownloadTemplateWrapper = styled(Box)({
  display: 'flex',
  alignItems: 'center',
  marginTop: '24px',
  cursor: 'pointer',
});
