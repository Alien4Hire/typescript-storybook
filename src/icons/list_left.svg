<svg id="EXPORT" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><defs><style>.cls-1{fill:#080808;}</style></defs><title>list_left</title><rect x="8" y="3" width="13" height="3"/><rect x="3" y="3" width="3" height="3"/><rect x="3" y="10" width="3" height="3"/><rect x="3" y="17" width="3" height="3"/><rect x="8" y="10" width="13" height="3"/><rect x="8" y="17" width="13" height="3"/></svg>