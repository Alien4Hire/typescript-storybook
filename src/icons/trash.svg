<svg id="EXPORT" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><defs><style>.cls-1{fill:#080808;}</style></defs><title>trash</title><path class="cls-1" d="M6,21a1,1,0,0,0,1,1H17a1,1,0,0,0,1-1V7H6Z"/><path class="cls-1" d="M18,3H14V2H9.92V3H6A1,1,0,0,0,5,4V6H19V4A1,1,0,0,0,18,3Z"/></svg>