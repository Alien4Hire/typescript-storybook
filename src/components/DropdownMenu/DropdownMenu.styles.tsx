import { ListItemText } from '@mui/material';
import { styled } from '@mui/material/styles';

export const ListItemTextContainer = styled(ListItemText)(
  ({ option, color }) => ({
    color: `${color} !important`,
    marginLeft: option.icon ? 0 : 36,
    paddingLeft: 16 * (option?.step || 0),
  })
);
