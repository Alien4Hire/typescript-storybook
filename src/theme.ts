import { createTheme } from '@mui/material/styles';
import mlColors from './util/ml-cdl-colors';
import {
  computerSmallMonitorMax,
  mobileScreenMax,
  mobileScreenMin,
  tabletScreenMax,
} from './util/breakpoints';

declare module '@mui/material/Typography' {
  interface TypographyPropsVariantOverrides {
    title: true;
    textLarge: true;
    textBasic: true;
    textSmall: true;
    textSub: true;
    buttonMedium: true;
    buttonSmall: true;
    buttonLarge: true;
  }
}

//UX variants definitions can be found at https://www.figma.com/file/bEonWYX4lesmqlFY6MqWCt/Course-Assembly-MVP?node-id=5540%3A249648
//Custom created variant names for typographies

declare module '@mui/material/styles/createTypography' {
  interface Typography {
    formTitle?: React.CSSProperties;
    formSubtitle?: React.CSSProperties;
    appTitle?: React.CSSProperties;
    title?: React.CSSProperties;
    textLarge?: React.CSSProperties;
    textBasic?: React.CSSProperties;
    textSmall?: React.CSSProperties;
    textSub?: React.CSSProperties;
    buttonMedium?: React.CSSProperties;
    buttonSmall?: React.CSSProperties;
    buttonLarge?: React.CSSProperties;
  }

  // allow configuration using `createMuiTheme`
  interface TypographyOptions {
    title?: React.CSSProperties;
    textLarge?: React.CSSProperties;
    textBasic?: React.CSSProperties;
    textSmall?: React.CSSProperties;
    textSub?: React.CSSProperties;
    buttonMedium?: React.CSSProperties;
    buttonSmall?: React.CSSProperties;
    buttonLarge?: React.CSSProperties;
  }
}

declare module '@mui/material/styles' {
  interface Theme {}
  interface ThemeOptions {}
  interface Palette {
    specials: Palette['primary'];
    aqua: Palette['primary'];
    black: Palette['primary'];
    blue: Palette['primary'];
    gray: Palette['primary'];
    green: Palette['primary'];
    magenta: Palette['primary'];
    orange: Palette['primary'];
    purple: Palette['primary'];
    red: Palette['primary'];
    teal: Palette['primary'];
    white: Palette['primary'];
    yellow: Palette['primary'];
  }
  interface PaletteOptions {
    specials: PaletteOptions['primary'];
    aqua: PaletteOptions['primary'];
    black: PaletteOptions['primary'];
    blue: PaletteOptions['primary'];
    gray: PaletteOptions['primary'];
    green: PaletteOptions['primary'];
    magenta: PaletteOptions['primary'];
    orange: PaletteOptions['primary'];
    purple: PaletteOptions['primary'];
    red: PaletteOptions['primary'];
    teal: PaletteOptions['primary'];
    white: PaletteOptions['primary'];
    yellow: PaletteOptions['primary'];
  }

  interface PaletteColor {
    macmillanRed?: string;
    offWhite?: string;
    veryLight?: string;
    medium?: string;
    veryDark?: string;
    fullPageDialogShadow?: string;
  }

  interface SimplePaletteColorOptions {
    macmillanRed?: string;
    offWhite?: string;
    veryLight?: string;
    medium?: string;
    veryDark?: string;
    fullPageDialogShadow?: string;
  }
}

export const theme = createTheme({
  palette: {
    background: {
      default: mlColors.off_white,
      paper: mlColors.pure_white,
    },
    primary: {
      main: mlColors.aqua,
    },
    secondary: {
      main: mlColors.aqua_dark,
    },
    specials: {
      main: mlColors.macmillan_red,
      macmillanRed: mlColors.macmillan_red,
      fullPageDialogShadow: 'rgba(0, 0, 0, 0.5)',
    },
    aqua: {
      main: mlColors.aqua,
      veryLight: mlColors.aqua_very_light,
      light: mlColors.aqua_light,
      medium: mlColors.aqua_medium,
      dark: mlColors.aqua_dark,
      veryDark: mlColors.dark_navy,
    },
    black: {
      main: mlColors.black,
    },
    blue: {
      main: mlColors.blue,
      veryLight: mlColors.blue_very_light,
      light: mlColors.blue_light,
      medium: mlColors.blue_medium,
      dark: mlColors.blue_dark,
    },
    gray: {
      main: mlColors.gray,
      veryLight: mlColors.gray_very_light,
      light: mlColors.gray_light,
      medium: mlColors.gray_medium,
      dark: mlColors.gray_dark,
      veryDark: mlColors.gray_very_dark,
    },
    green: {
      main: mlColors.green,
      veryLight: mlColors.green_very_light,
      light: mlColors.green_light,
      medium: mlColors.green_medium,
      dark: mlColors.green_dark,
    },
    magenta: {
      main: mlColors.magenta,
      veryLight: mlColors.magenta_very_light,
      light: mlColors.magenta_light,
      medium: mlColors.magenta_medium,
      dark: mlColors.magenta_dark,
    },
    orange: {
      main: mlColors.orange,
      veryLight: mlColors.orange_very_light,
      light: mlColors.orange_light,
      medium: mlColors.orange_medium,
      dark: mlColors.orange_dark,
    },
    purple: {
      main: mlColors.purple,
      veryLight: mlColors.purple_very_light,
      light: mlColors.purple_light,
      medium: mlColors.purple_medium,
      dark: mlColors.purple_dark,
    },
    red: {
      main: mlColors.red,
      veryLight: mlColors.red_very_light,
      light: mlColors.red_light,
      medium: mlColors.red_medium,
      dark: mlColors.red_dark,
    },
    teal: {
      main: mlColors.teal,
      veryLight: mlColors.teal_very_light,
      light: mlColors.teal_light,
      medium: mlColors.teal_medium,
      dark: mlColors.teal_dark,
      veryDark: mlColors.teal_very_dark,
    },
    white: {
      main: mlColors.pure_white,
      offWhite: mlColors.off_white,
    },
    yellow: {
      main: mlColors.yellow,
      veryLight: mlColors.yellow_very_light,
      light: mlColors.yellow_light,
      medium: mlColors.yellow_medium,
      dark: mlColors.yellow_dark,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: mobileScreenMin,
      md: mobileScreenMax,
      lg: tabletScreenMax,
      xl: computerSmallMonitorMax,
    },
  },
  spacing: 8,
  typography: {
    fontSize: 14,
    fontFamily: [
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
      'Oxygen',
      'Ubuntu',
      'Cantarell',
      '"Fira Sans"',
      '"Droid Sans"',
      '"Helvetica Neue"',
      'Source Sans Pro',
      'sans-serif',
    ].join(','),
    allVariants: {
      fontWeight: 'normal',
    },
    button: {
      fontSize: 14,
      textTransform: 'uppercase',
    },
    h1: {
      color: mlColors.black,
      fontSize: 32,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h2: {
      color: mlColors.black,
      fontSize: 24,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h3: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    h4: {
      color: mlColors.black,
      fontSize: 16,
      fontFamily: 'Source Sans Pro',
      fontWeight: 600,
    },
    title: {
      color: mlColors.black,
      fontSize: 28,
      fontFamily: 'Source Sans Pro',
      fontWeight: 200,
      fontStyle: 'normal',
    },
    textLarge: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textBasic: {
      color: mlColors.black,
      fontSize: 16,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textSmall: {
      color: mlColors.black,
      fontSize: 14,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    textSub: {
      color: mlColors.black,
      fontSize: 13,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonLarge: {
      color: mlColors.black,
      fontSize: 15,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonMedium: {
      color: mlColors.black,
      fontSize: 18,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
    buttonSmall: {
      color: mlColors.black,
      fontSize: 12,
      fontFamily: 'Source Sans Pro',
      fontWeight: 400,
      fontStyle: 'normal',
    },
  },
  components: {
    MuiAlert: {
      styleOverrides: {
        filledError: {
          background: mlColors.red_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.red}`,
        },
        filledInfo: {
          background: mlColors.blue_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.blue}`,
        },
        filledSuccess: {
          background: mlColors.green_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.green}`,
        },
        filledWarning: {
          background: mlColors.yellow_very_light,
          color: mlColors.black,
          border: `1px solid ${mlColors.yellow}`,
        },
      },
    },
    // MuiButton: {
    //   styleOverrides: {
    //     root: {
    //       fontSize: '1rem',
    //       textTransform: 'none',
    //       height: '35px',
    //       padding: '0 15px',
    //       margin: '5px',
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #FFFFFF',
    //       },
    //       '&:disabled': {
    //         background: '#c8c8c8',
    //         color: '#f3f3f3',
    //       },
    //     },
    //     textSecondary: {
    //       fontSize: '0.9375rem',
    //       height: '34px',
    //       lineHeight: '33px',
    //       padding: '0 15px',
    //       borderRadius: '3px',
    //       '-webkit-border-radius': '3px',
    //       '-ms-border-radius': '3px',
    //       '-moz-border-radius': '3px',
    //       textDecoration: 'none',
    //       color: '#00758e',
    //       cursor: 'pointer',
    //       background: 'transparent',
    //       border: '1px solid #00758e',
    //       overflow: 'hidden',
    //     },
    //     containedPrimary: {
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #FFFFFF',
    //       },
    //       '&:disabled': {
    //         background: '#c8c8c8',
    //         color: '#f3f3f3',
    //       },
    //     },
    //     outlinedSecondary: {
    //       '&:focus': {
    //         boxShadow: '0 0 6px rgba(0, 117, 142, 1)',
    //         border: '1px solid #00758e',
    //       },
    //       '&:disabled': {
    //         color: '#989898',
    //         fill: '#c8c8c8',
    //         backgroundColor: '#f3f3f3',
    //       },
    //     },
    //   },
    // },
    MuiButtonBase: {
      defaultProps: {
        disableRipple: true, // No more ripple, on the whole application!
        focusRipple: false,
      },
    },
  },
});
