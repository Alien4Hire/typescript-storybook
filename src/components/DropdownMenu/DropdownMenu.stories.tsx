import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';
import MoreVertIcon from '@mui/icons-material/MoreVert';

import IconButton from '@mui/material/IconButton';
import { DropdownMenu } from './DropdownMenu';
import { ReactComponent as TrashIcon } from '../../icons/trash.svg';
import { ReactComponent as ListLeftIcon } from '../../icons/list_left.svg';

export default {
  title: 'Component/DropdownMenu',
  component: DropdownMenu,
} as ComponentMeta<typeof DropdownMenu>;

const Template: ComponentStory<typeof DropdownMenu> = () => {
  const options = [
    {
      label: 'Remove from Resource Structure',
      value: 'idaosnaisonf',
      icon: <TrashIcon />,
    },
    {
      label: 'Show Location in Structure',
      value: 'sodnfois',
      icon: <ListLeftIcon />,
    },
    {
      label: 'Apply Activity Setting',
      value: 'asofpafpnapon',
    },
  ];
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);

  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleItemClick = (selectedValue: string) => {
    handleClose();
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  return (
    <div>
      <IconButton aria-label="more" aria-haspopup onClick={handleClick}>
        <MoreVertIcon />
      </IconButton>
      <DropdownMenu
        open={Boolean(anchorEl)}
        anchorEl={anchorEl}
        options={options}
        onItemClick={handleItemClick}
        onClose={handleClose}
      />
    </div>
  );
};

export const DropDownMenuView = Template.bind({});
