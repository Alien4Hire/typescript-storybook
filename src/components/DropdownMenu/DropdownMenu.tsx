import { ReactElement } from 'react';

import { Menu, MenuItem, MenuProps, ListItemIcon } from '@mui/material';
import { theme } from '../../theme';
import { ListItemTextContainer } from './DropdownMenu.styles';

export type MlDropdownMenuOptionPropsT = {
  label: string;
  value: string;
  icon?: ReactElement;
  step?: number;
};

export type MlDropdownMenuPropsT = MenuProps & {
  options: MlDropdownMenuOptionPropsT[];
  labelColor?: string;
  testId?: string;
  onItemClick: (e: string) => void;
  anchorEl: HTMLElement | null;
  onClose: () => void;
};

const ITEM_HEIGHT = 48;

export const DropdownMenu = ({
  open,
  options,
  labelColor = theme.palette.black.main,
  testId,
  onItemClick,
  anchorEl,
  onClose,
}: MlDropdownMenuPropsT) => {
  return (
    <Menu
      data-testid={testId}
      MenuListProps={{
        'aria-labelledby': 'long-button',
      }}
      anchorEl={anchorEl}
      open={open}
      onClose={onClose}
      PaperProps={{
        style: {
          maxHeight: ITEM_HEIGHT * 4.5,
        },
      }}
    >
      {options.map((option) => (
        <MenuItem
          key={option.value}
          selected={option?.value === 'Pyxis'}
          onClick={(e) => {
            e.stopPropagation();
            onItemClick(option.value);
            onClose();
          }}
        >
          {option.icon && <ListItemIcon>{option.icon}</ListItemIcon>}
          <ListItemTextContainer option={option} color={labelColor}>
            {option.label}
          </ListItemTextContainer>
        </MenuItem>
      ))}
    </Menu>
  );
};
