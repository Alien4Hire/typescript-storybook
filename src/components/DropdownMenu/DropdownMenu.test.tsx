import { render, fireEvent, screen } from '@testing-library/react';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { DropdownMenu } from './DropdownMenu';
import { theme } from '../../theme';
import '../../setupTests';
import { ReactComponent as TrashIcon } from '../../icons/trash.svg';
import { ReactComponent as ListLeftIcon } from '../../icons/list_left.svg';

const baseProps = {
  open: true,
  options: [
    {
      label: 'Remove from Resource Structure',
      value: 'idaosnaisonf',
      icon: <TrashIcon />,
    },
    {
      label: 'Show Location in Structure',
      value: 'sodnfois',
      icon: <ListLeftIcon />,
    },
    {
      label: 'Apply Activity Setting',
      value: 'asofpafpnapon',
    },
  ],
  onItemClick: jest.fn(),
  anchorEl: document.createElement('div'),
  onClose: jest.fn(),
};

const renderComponent = (customProps?: {
  testId?: string;
  onItemClick?: jest.Mock<any, any>;
  onClose?: jest.Mock<any, any>;
}) => {
  const props = { ...baseProps, ...customProps };
  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={theme}>
        <DropdownMenu {...props} />
      </ThemeProvider>
    </StyledEngineProvider>
  );
};

describe('<DropdownMenu />', () => {
  it('renders without crashing', () => {
    const view = renderComponent();
    expect(view).toMatchSnapshot();
  });

  it('should have a data-test-id', () => {
    const testId = 'test-dropdown-menu';
    renderComponent({ testId });
    const element = screen.getByTestId(testId);
    expect(element).toBeInTheDocument();
  });

  it('should fire an event when a item gets clicked', () => {
    const mockItemClick = jest.fn();
    renderComponent({ onItemClick: mockItemClick });
    const element = screen.getByText('Remove from Resource Structure');
    fireEvent.click(element);
    expect(mockItemClick).toHaveBeenCalledTimes(1);
    expect(mockItemClick).toHaveBeenCalledWith('idaosnaisonf');
  });

  it('should fire the onClose event', () => {
    const mockCloseClick = jest.fn();
    renderComponent({ onClose: mockCloseClick });
    const element = screen.getByText('Remove from Resource Structure');
    fireEvent.click(element);
    expect(mockCloseClick).toHaveBeenCalledTimes(1);
  });
});
