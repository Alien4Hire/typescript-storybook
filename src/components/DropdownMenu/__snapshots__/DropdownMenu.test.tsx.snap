// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownMenu /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c5 {
  background-color: #ffffff;
  color: rgba(0,0,0,0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.c3 {
  background-color: #ffffff;
  color: rgba(0,0,0,0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4,0,0.2,1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c9::-moz-focus-inner {
  border-style: none;
}

.c9.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c14 {
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  display: block;
}

.c1 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: transparent;
  -webkit-tap-highlight-color: transparent;
}

.c0 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
}

.c2 {
  z-index: -1;
}

.c7 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c11 {
  min-width: 56px;
  color: rgba(0,0,0,0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c12 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.c6 {
  position: absolute;
  overflow-y: auto;
  overflow-x: hidden;
  min-width: 16px;
  min-height: 16px;
  max-width: calc(100% - 32px);
  max-height: calc(100% - 32px);
  outline: 0;
}

.c4 {
  max-height: calc(100% - 96px);
  -webkit-overflow-scrolling: touch;
}

.c8 {
  outline: 0;
}

.c10 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen,Ubuntu,Cantarell,"Fira Sans","Droid Sans","Helvetica Neue",Source Sans Pro,sans-serif;
  font-weight: normal;
  font-size: 1rem;
  line-height: 1.5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 48px;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
  white-space: nowrap;
  padding-left: 16px;
  padding-right: 16px;
}

.c10:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0,0,0,0.04);
}

.c10.Mui-selected {
  background-color: rgba(0,117,142,0.08);
}

.c10.Mui-selected.Mui-focusVisible {
  background-color: rgba(0,117,142,0.2);
}

.c10.Mui-selected:hover {
  background-color: rgba(0,117,142,0.12);
}

.c10.Mui-focusVisible {
  background-color: rgba(0,0,0,0.12);
}

.c10.Mui-disabled {
  opacity: 0.38;
}

.c10 + .MuiDivider-root {
  margin-top: 8px;
  margin-bottom: 8px;
}

.c10 + .MuiDivider-inset {
  margin-left: 52px;
}

.c10 .MuiListItemText-root {
  margin-top: 0;
  margin-bottom: 0;
}

.c10 .MuiListItemText-inset {
  padding-left: 36px;
}

.c10 .MuiListItemIcon-root {
  min-width: 36px;
}

.c13 {
  color: #080808 !important;
  margin-left: 0;
  padding-left: 0;
}

.c15 {
  color: #080808 !important;
  margin-left: 36px;
  padding-left: 0;
}

@media print {
  .c9 {
    color-adjust: exact;
  }
}

@media (hover:none) {
  .c10:hover {
    background-color: transparent;
  }
}

@media (hover:none) {
  .c10.Mui-selected:hover {
    background-color: rgba(0,117,142,0.08);
  }
}

@media (min-width:660px) {
  .c10 {
    min-height: auto;
  }
}

<body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="c0 MuiModal-root MuiPopover-root-kLYbPy MuiPopover-root MuiMenu-root-egkglY MuiMenu-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="c1 MuiBackdrop-root MuiBackdrop-invisible c2"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="c3 MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c4 MuiPaper-root-jrxuSB c5 MuiMenu-paper MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 c6 MuiPopover-paper"
        style="opacity: 1; transform: scale(1, 1); max-height: 216px; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
        tabindex="-1"
      >
        <ul
          aria-labelledby="long-button"
          class="c7 MuiList-root MuiList-padding c8 MuiMenu-list"
          role="menu"
          tabindex="-1"
        >
          <li
            class="c9 MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root Mui-focusVisible c10"
            role="menuitem"
            tabindex="0"
          >
            <div
              class="c11 MuiListItemIcon-root"
            >
              <svg>
                trash.svg
              </svg>
            </div>
            <div
              class="c12 MuiListItemText-root c13"
              color="#080808"
              option="[object Object]"
            >
              <span
                class="c14 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
              >
                Remove from Resource Structure
              </span>
            </div>
          </li>
          <li
            class="c9 MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root c10"
            role="menuitem"
            tabindex="-1"
          >
            <div
              class="c11 MuiListItemIcon-root"
            >
              <svg>
                list_left.svg
              </svg>
            </div>
            <div
              class="c12 MuiListItemText-root c13"
              color="#080808"
              option="[object Object]"
            >
              <span
                class="c14 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
              >
                Show Location in Structure
              </span>
            </div>
          </li>
          <li
            class="c9 MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root c10"
            role="menuitem"
            tabindex="-1"
          >
            <div
              class="c12 MuiListItemText-root c15"
              color="#080808"
              option="[object Object]"
            >
              <span
                class="c14 MuiTypography-root MuiTypography-body1 MuiListItemText-primary"
              >
                Apply Activity Setting
              </span>
            </div>
          </li>
        </ul>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": @media print {

}

@media (hover:none) {

}

@media (hover:none) {

}

@media (min-width:660px) {

}

<div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
